packages/product-react/src/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,217 @@
+    setActiveTab(focusedTab)();
+  }, [focusedTab]);
+
+  if (tabs.length === 0) {
+    return null;
+  }
+
+  return (
+    <TabContext.Provider value={{ selectedTab: selectedTabId, tabs }}>
+      <div className="ray-tabs__wrapper" role="tablist">
+        {tabs.map(tab => {
+          const tabIsActive = tab.id === selectedTabId;
+
+          return (
+            <div
+              key={tab.id}
+              className={classnames('ray-tabs__button-wrapper', {
+                'ray-tabs__button-wrapper--active': tabIsActive
+              })}
+            >
+              <Button
+                type="tertiary"
+                aria-label={tab.ariaLabel}
+                aria-selected={tabIsActive}
+                onClick={setActiveTab(tab.id)}
+                role="tab"
+                aria-controls={tab.id}
+                ref={tabRefs[tab.id]}
+                tabIndex={tabIsActive ? 0 : -1}
+                data-testid="tab-button"
+                onFocus={onButtonFocus}
+                onBlur={onButtonBlur}
+                compact={compact}
+              >
+                {tab.label}
+              </Button>
+            </div>
+          );
+        })}
+      </div>
+      {props.children}
+    </TabContext.Provider>
+  );
+};
+
+Tabs.displayName = 'Tabs';
+
+export default Tabs;
+
+interface IFocusManager {
+  tabs: ITab[];
+  firstFocusedId: string;
+  buttonsFocused: boolean;
+}
+
+function useFocusManager(props: IFocusManager) {
+  const { tabs, firstFocusedId, buttonsFocused } = props;
+  const [focusedTab, setFocusedTab] = React.useState(firstFocusedId);
+  const [isMounted, setIsMounted] = React.useState(false);
+
+  const tabRefs: ITabRefs = tabs.reduce((acc, { id }) => {
+    return { ...acc, [id]: React.useRef<HTMLButtonElement>(null) };
+  }, {});
+
+  React.useEffect(() => {
+    if (!isMounted) {
+      return setIsMounted(true);
+    }
+
+    const tabRef = tabRefs[focusedTab];
+
+    if (tabRef.current) {
+      tabRef.current.focus();
+    }
+  }, [focusedTab]);
+
+  const focusFirstTab = () => {
+    const firstTabId = getIdAt(tabs, 0);
+
+    setFocusedTab(firstTabId);
+  };
+
+  const focusLastTab = () => {
+    const lastIndex = getLastIndex(tabs);
+    const lastTabId = getIdAt(tabs, lastIndex);
+
+    setFocusedTab(lastTabId);
+  };
+
+  const focusNextTab = () => {
+    setFocusedTab(currentFocused => {
+      const currentIndex = getTabIndex(tabs, currentFocused);
+
+      if (currentIndex === getLastIndex(tabs)) {
+        return getFirstTabId(tabs);
+      }
+
+      return getIdAt(tabs, currentIndex + 1);
+    });
+  };
+
+  const focusPreviousTab = () => {
+    setFocusedTab(currentFocused => {
+      const currentIndex = getTabIndex(tabs, currentFocused);
+
+      if (currentIndex === 0) {
+        return getLastTabId(tabs);
+      }
+
+      return getIdAt(tabs, currentIndex - 1);
+    });
+  };
+
+  const focusRoving = ({ key }: { key: string }) => {
+    switch (key) {
```

Comment:
should there be a `default` case to unfocus?

if the user pressed `tab` it should focus something else on the page

Reply:
No need to handle that in here, that should happen just by browser default.